sdk/storage/azblob/blockblob/mmf_unix.go

```diff
@@ -0,0 +1,36 @@
+//go:build go1.18 && (linux || darwin || freebsd || openbsd || netbsd)
+// +build go1.18
+// +build linux darwin freebsd openbsd netbsd
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package blockblob
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+// mmb is a memory mapped buffer
+type mmb []byte
+
+// newMMB creates a new memory mapped buffer with the specified size
+func newMMB(size int64) (mmb, error) {
+	prot, flags := syscall.PROT_READ|syscall.PROT_WRITE, syscall.MAP_ANONYMOUS|syscall.MAP_SHARED
+	addr, err := syscall.Mmap(-1, 0, int(size), prot, flags)
+	if err != nil {
+		return nil, os.NewSyscallError("Mmap", err)
+	}
+	return mmb(addr), err
+}
+
+// delete cleans up the memory mapped buffer
+func (m *mmb) delete() {
+	err := syscall.Munmap(*m)
+	*m = nil
+	if err != nil {
+		panic(fmt.Sprintf("fatal error unmapping mmb: %v", err))
```

Comment:
Bold having panic here. I'm OK with this.

Reply:
This was copied from track 1.5.  I'm on the fence about this, but I don't know how "fatal" failing to unmap the buffer is.  I suppose if we surface this as an `os.SyscallError` then callers _should_ be savvy enough to treat this as fatal.

Reply:
Also, it makes it hard to reason about the API.  If we returned an error in this case, but the blob was uploaded, how is a customer supposed to figure this out?

Reply:
For the record, I'd also be OK with _ = syscall.Mumap(*m) and just throw the error away. In Win32 no one ever checked the true/false return from calling CloseHandle.

Reply:
I suppose that in the event this happens, it manifests as a memory leak and wouldn't cause actual memory corruption.  Perhaps it would be better to just log that this happened and move on instead.

Reply:
How would we log it? I think we either keep the panic or make this method return an error (just like how Win32's CloseHandle returns a Boolean). The caller can then decide to propogate it or ignore it.

Reply:
Through our SDK logging facility.  I already made this change, take a look to see what you think.

Reply:
I can switch it to returning an error, but before I do that, we need to decide how to message this to the caller.  In this case, the upload succeeded, we just failed to clean up the buffer(s).  And since customers pay for data transfers, I think it would be disingenuous to present the error in such a way that they think they'd need to redo the upload.  So, we'd have this corner-case where the caller receives an `os.SyscallError` _but_ the operation actually succeeded.

Reply:
You've convinced me that `panic` is the best choice for now, I've switched it back.